codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/Waiters.java

```diff
@@ -0,0 +1,835 @@
+            w.write("var remainingTime = m.maxWaitTime");
+
+            // start loop
+            w.write("for {");
+
+            w.writeDocs("check number of attempts");
+            w.write("if m.maxAttempts == attempt { break }");
+            w.write("");
+            w.write("attempt++");
+            w.write("");
+
+            w.write("attemptInput := in");
+            w.write("attemptInput.Request = m.requestCloner(attemptInput.Request)");
+
+            w.write("");
+
+            // start retry only behavior
+            w.write("if attempt > 1 {");
+
+            Symbol computeDelaySymbol = SymbolUtils.createValueSymbolBuilder(
+                    "ComputeDelay", SmithyGoDependency.SMITHY_WAITERS
+            ).build();
+            w.writeDocs("compute exponential backoff between waiter retries");
+            w.write("delay = $T(m.minDelay, m.maxDelay, remainingTime, attempt)", computeDelaySymbol);
+            w.write("");
+
+            w.writeDocs("update the remaining time");
+            w.write("remainingTime = remainingTime - delay");
+            w.write("");
+
+            w.writeDocs("rewind transport stream");
+            w.write("if rewindable, ok := in.Request.(interface{ RewindStream() error }); ok {");
+            w.write("if err := rewindable.RewindStream(); err != nil {");
+            w.addUseImports(SmithyGoDependency.FMT);
+            w.write("return out, metadata, fmt.Errorf(\"failed to rewind transport stream for retry, %w\", err)\n}");
+            w.write("}");
+            w.write("");
+
+            Symbol sleepWithContextSymbol = SymbolUtils.createValueSymbolBuilder(
+                    "SleepWithContext", SmithyGoDependency.SMITHY_WAITERS
+            ).build();
+            w.writeDocs("sleep for the delay amount before invoking a request");
+            w.openBlock("if err = $T(ctx, delay); err != nil {", "}",
+                    sleepWithContextSymbol, () -> {
+                        w.write("return out, metadata, fmt.Errorf(\"request cancelled while waiting, %w\", err)");
+                    });
+
+            // enable logger
+            w.writeDocs("log attempt");
+            w.openBlock("if m.enableLogger {", "}", () -> {
+                w.write("logger.Logf(logging.Debug, "
+                        + "fmt.Sprintf(\"retrying waiter request, attempt count: %d\", attempt))\n");
+            });
+            w.write("");
+            // end retry only behavior
+            w.write("}");
+
+            w.write("");
+            w.writeDocs("attempt request");
+            w.write("out, metadata, err = next.HandleFinalize(ctx, attemptInput)");
+            w.write("");
+
+            w.writeDocs("check for state mutation");
+            w.write("output := out.Result.($P)", outputSymbol);
+            w.write("retryable, err := m.waiterStateMutator(ctx, m.params, output, err)");
+            w.write("");
+
+            w.openBlock("if !retryable || err != nil {", "}", () -> {
```

Comment:
nit not really anything being mutated, but checking if retryable.

also don't really need the extra `write("")` before the error check so they are